meson.build

```
@@ -0,0 +1,494 @@
+project('vips', 'c', 'cpp',
+    version: '8.13.0',
+    meson_version: '>=0.56',
+    default_options: [
+        'c_std=c99'
+    ]
+)
+
+version_parts = meson.project_version().split('.')
+version_major = version_parts[0]
+version_minor = version_parts[1]
+version_patch = version_parts[2]
+
+# rules:
+# sources changed: increment revision
+# binary interface changed: increment current, reset revision to 0
+#   binary interface changes backwards compatible?: increment age
+#   binary interface changes not backwards compatible?: reset age to 0
+library_current = 57
+library_age = 15
+library_revision = 0
+library_version = '@0@.@1@.@2@'.format(library_current - library_age, library_age, library_revision)
+
+gnome = import('gnome')
+pymod = import('python')
+pkg = import('pkgconfig')
+i18n = import('i18n')
+
+add_project_arguments('-Drestrict=__restrict', language: 'cpp')
+
+cc = meson.get_compiler('c')
+
+glib_dep = dependency('glib-2.0')
+gio_dep = dependency('gio-2.0')
+gobject_dep = dependency('gobject-2.0')
+gmodule_dep = dependency('gmodule-no-export-2.0')
+expat_dep = dependency('expat')
+m_dep = cc.find_library('m', required : false)
+
+libvips_deps = [
+    glib_dep,
+    gio_dep,
+    gobject_dep,
+    gmodule_dep,
+    expat_dep,
+    m_dep,
+]
+
+prefix_dir = get_option('prefix')
+lib_dir = prefix_dir / get_option('libdir')
+if gmodule_dep.get_variable(pkgconfig: 'gmodule_supported') != 'false'
+    modules_enabled = get_option('modules').enabled() or get_option('modules').auto()
+elif get_option('modules').enabled()
+    error('GModule is not supported on your system, please reconfigure with -Dmodules=disabled')
+else
+    modules_enabled = false
+endif
+
+module_dir = lib_dir / 'vips-modules-@0@.@1@'.format(version_major, version_minor)
+
+cfg_var = configuration_data()
+cfg_var.set_quoted('G_LOG_DOMAIN', 'VIPS')
+
+if (glib_dep.version().version_compare('>=2.48'))
+    cfg_var.set('HAVE_CHECKED_MUL', '1')
+endif
+if (glib_dep.version().version_compare('>=2.62'))
+    cfg_var.set('HAVE_DATE_TIME_FORMAT_ISO8601', '1')
+endif
+
+if modules_enabled
+    cfg_var.set('ENABLE_MODULES', '1')
+endif
+
+func_names = [ 'getcwd', 'gettimeofday', 'getwd', 'memset', 'munmap', 'putenv', 'realpath', 'strcasecmp', 'strchr', 'strcspn', 'strdup', 'strerror', 'strrchr', 'strspn', 'vsnprintf', 'realpath', 'mkstemp', 'mktemp', 'random', 'rand', 'sysconf', 'atexit', '_aligned_malloc', 'posix_memalign', 'memalign', 'cbrt', 'hypot', 'atan2', 'asinh' ]
```

Note to myself: most checks are currently unused, this can be simplified to:
```suggestion
func_names = [ 'vsnprintf', 'atexit', '_aligned_malloc', 'posix_memalign', 'memalign', 'cbrt', 'hypot', 'atan2', 'asinh' ]
```